apps/etterlatte-behandling/src/main/kotlin/behandling/omregning/OmregningKlassifikasjonskodeJobService.kt

```diff
@@ -0,0 +1,136 @@
+                    revurderingaarsak = Revurderingaarsak.OMREGNING,
+                    fradato = foersteVirkningstidspunktForSak,
+                    verifiserUtbetalingUendret = true,
+                )
+
+            logger.info("Publiserer omregningshendelse for sak $sakId på kafka")
+            publiserHendelse(omregningData)
+        }
+    }
+
+    private fun verifiserAtFoerstegangsbehandlingErIverksatt(foerstegangsbehandling: Foerstegangsbehandling) {
+        logger.info("Verifiserer at førstegangsbehandling er iverksatt")
+        if (foerstegangsbehandling.status != BehandlingStatus.IVERKSATT) {
+            throw InternfeilException(
+                "Omregning feilet for førstegangsbehandling ${foerstegangsbehandling.id} er ikke iverksatt (status=${foerstegangsbehandling.status})",
+            )
+        }
+    }
+
+    private fun verifiserAtIngenPabegynteBehandlingerFinnes(
+        sakId: SakId,
+        behandlinger: List<Behandling>,
+    ) {
+        logger.info("Verifiserer at ingen behandlinger er under arbeid")
+        behandlinger.forEach {
+            if (it.status !in listOf(BehandlingStatus.IVERKSATT, BehandlingStatus.AVBRUTT, BehandlingStatus.AVSLAG)) {
+                throw InternfeilException(
+                    "Omregning feilet fordi sak $sakId har en behandling ${it.id} med status ${it.status} - dette er ikke tillatt ved omregning",
+                )
+            }
+        }
+    }
+
+    private fun hentFoersteVirkningstidspunktForSak(sakId: SakId): LocalDate {
+        val foerstegangsbehandling = behandlingService.hentFoerstegangsbehandling(sakId)
+        val foersteVirkningstidspunktForSak =
+            foerstegangsbehandling.virkningstidspunkt?.dato?.atDay(1)
+                ?: throw InternfeilException("Behandling ${foerstegangsbehandling.id} manglet virkningstidspunkt")
+        return foersteVirkningstidspunktForSak
+    }
+
+    private fun publiserHendelse(omregningData: OmregningData) {
+        val correlationId = getCorrelationId()
+        val key = omregningData.sakId.toString()
+        val eventType = OmregningHendelseType.KLAR_FOR_OMREGNING.lagEventnameForType()
+
+        val params =
+            mapOf(
+                SAK_ID_KEY to omregningData.sakId,
+                TEKNISK_TID_KEY to LocalDateTime.now(),
+                CORRELATION_ID_KEY to correlationId,
+                HENDELSE_DATA_KEY to omregningData.toPacket(),
+            )
+
+        val (partition, offset) = kafkaProdusent.publiser(key, JsonMessage.newMessage(eventType, params).toJson())
+
+        logger.info(
+            "Hendelse med eventType $eventType for sak ${omregningData.sakId} publisert på kafka med partition $partition, " +
+                "offset $offset og correlationId $correlationId",
+        )
+    }
+
+    companion object {
+        val kjoering = "ENDRE_KLASSIFIKASJONSKODE_FOR_PERIODER_2023"
+        val antall = 1
+        val ekskluderteSaker = emptyList<Long>() // TODO
```

Comment:
Er tanken å endre disse, bygge og redeploye behandling-appen hvis man støter på problemer med noen saker eller ønsker å endre antallet? Kanskje burde vi ha et opplegg for å kjøre en engangs-app for slike engangsjobber?

Reply:
Ja, er litt styr de greiene her. Burde nok se på bedre løsninger for å få satt i gang slike ting. Problemet er at de vanligvis er litt ulike så er vanskelig å finne noe som passer alle de ulike casene vi kommer ut for. Men absolutt noe vi kan tenke litt rundt nå som vi har hatt en del ulike use-cases på dette.